src/libraries/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -226,9 +226,9 @@ public static bool TryParse([NotNullWhen(true)] string? ipString, [NotNullWhen(t
             return (address != null);
         }
 
-        public static bool TryParse(ReadOnlySpan<char> ipSpan, [NotNullWhen(true)] out IPAddress? address)
+        public static bool TryParse(ReadOnlySpan<char> ipString, [NotNullWhen(true)] out IPAddress? address)
```

Comment:
@dotnet/ncl this looks like less desirable than what was previously here.  If you intended to change the reference assembly prameter name, take that through API review and file a breaking change issue.

Reply:
Original API proposal: https://github.com/dotnet/runtime/issues/22918
` public static bool TryParse(ReadOnlySpan<char> ipChars, out IPAddress address);`

According to the original issue, it should be `ipChars`. The code is quite old (merged 3 years ago on Aug 22, 2017). @karelz is parameter name change breaking change? If not, should we align it with the original API proposal?

Reply:
The publicly documented API is `ipString` https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.tryparse?view=netcore-3.1#System_Net_IPAddress_TryParse_System_ReadOnlySpan_System_Char__System_Net_IPAddress__

It's also what we've shipped since 2.1 https://github.com/dotnet/corefx/commit/0029b322154053a76b085ccb6f3f820a5c9a263f#diff-08121b9ae82947c9c21c6bb7c18202a4R223

> is parameter name change breaking change?

It's source breaking when source uses named parameters. EG:
```
        ReadOnlySpan<char> ip = "1.1.1.1";
        IPAddress res;
        IPAddress.TryParse(ipString: ip, out res);
```
If you were to correct the ref, this source would break.

Technically changing implementation is also observable, through reflection, but we typically prioritize avoiding source breaking changes on these issues.

Reply:
@dotnet/ncl please weigh in on what you'd like to do here.

Reply:
cc: @karelz @scalablecory @wfurt

Reply:
We should keep the same name we had in previous release 3.1 IMO ... at this stage of 5.0, I would avoid discussions about changing param name.
How was the discrepancy introduced?

Reply:
In the original PR, just overlooked when doing review I guess.

Reply:
@karelz we're weighing which way is less breaking (fixing ref vs fixing src).  Currently we're thinking fixing ref is [less breaking](https://github.com/dotnet/runtime/pull/39741#issuecomment-663191615), and we still have a chance to do that.  Are you suggesting that instead we don't make any changes and keep ref and src out of sync?

Reply:
OK, the original PR https://github.com/dotnet/corefx/pull/23224 introduced the inconsistency: ref with `ipString`, but implementation with `ipSpan` (while approved API asked for `ipChars`).
I agree that changing the implementation to match the ref name makes most sense at this point -- it is the least breaking.

I don't think it is worth to make the breaking change of parameter name in ref in 5.0 (and likely never) to something else.

Reply:
(our replies crossed and I didn't see your summary in https://github.com/dotnet/runtime/pull/39741#issuecomment-663191615)

Honestly, it probably does not matter which way we go. I am fine to go with recommendation from API review folks. `ipSpan` is not ideal ... if API review group is fine with such arg name, we can use it. Otherwise, let's stick to `ipString`.
The risk of breaking customers is either way fairly small IMO.